### PR TITLE
Add command to check duplicates in translation catalog

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | This PR adds a new command `translation:find-duplicates` in order to display sames strings with different parameter names. These strings are currently being duplicated in the translation page, which means wasted time for everybody.
| Type?         | new feature
| Category?     | CO
| BC breaks?    | Nope
| Deprecations? | Nope
| Fixed ticket? | /
| How to test?  | `app/console prestashop:translation:find-duplicates`

capture du 2017-04-28 16-32-21

### CODE REVIEW

File: src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        $progress->finish();
+        $output->writeln('');
+
+        if (count($duplicates)) {
+            $output->writeln('Duplicates found:');
+            dump($duplicates);
+        }else {
+            $output->writeln('Awww yisss ! There is no duplicate in your translator catalogue.');
```
Comment (aleeks) [inline src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php]: 😄

**Review by aleeks — APPROVED** [review]
Looks good to me. 
Maybe can you add more documentation/comment ?

File: src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+class CheckTranslationDuplicatesCommand extends ContainerAwareCommand
+{
+
+    protected function configure()
+    {
+        $this
+            ->setName('translation:find-duplicates')
```
Comment (maximebiloe) [inline src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php]: Could you use the same prefix as our other commands, please? `prestashop:...`

**Comment by maximebiloe** [comment]
Nice job here!!!

**Comment by Quetzacoalt91** [comment]
The PR has been updated with the following changes:
* Command name updated
* PHP doc added
* Reuse regex available from `PrestaShopTranslatorTrait`
* Add values on return for Travis routine

File: src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        // If we have duplicates to fix, let's display them and return their count.
+        // This will allow us to add the command in the tests.
+        if (count($duplicates)) {
+            $output->writeln('Duplicates found:');
+            dump($duplicates);
+            return count($duplicates, true);
+        }
+
+        $output->writeln('Awww yisss ! There is no duplicate in your translator catalogue.');
```
Comment (xBorderie) [inline src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php]: No space before "!".
We use US English -> "catalog".

Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Command/CheckTranslationDuplicatesCommand.php]: Fixed!

**Comment by AlexEven** [comment]
Are we waiting for something special to merge this PR?
I kinda need it to clean the translation duplicates ^^;

Thanks!

**Comment by aleeks** [comment]
Thank you @Quetzacoalt91

**Pull request merged**